administrative-sdk/choice-challenge/choice-challenge.js

```diff
@@ -49,10 +52,25 @@ export default class ChoiceChallenge {
      * @type {string} [question] A hint or question related to the choices.
      */
     this.question = question;
-    if (typeof choices !== 'object') {
+
+    if (!Array.isArray(choices)) {
       throw new Error(
-        'choices parameter of type "Array" is required');
+        'choices parameter of type "string|object Array" is required');
+    }
+    if (choices.length === 0) {
+      throw new Error('non-empty choices parameter is required');
     }
+    choices.map(choice => {
+      if (typeof choice !== 'string' && typeof choice !== 'object') {
+        throw new Error(
+          'choices parameter of type "string|object Array" is required');
+      }
+
+      if (/\d/.test(choice)) {
```

Comment:
I'd leave validation to the API itself (if it doesn't already, I'd file an issue for that). These rules may change one day. It'll probably be modified in the API and the SDK might be forgotten.